packages/common/src/ui/components/buttons/SplitButton.tsx

```diff
@@ -6,109 +6,105 @@ import { ChevronDownIcon } from '../../icons';
 import { ButtonWithIcon, ButtonWithIconProps } from './standard/ButtonWithIcon';
 import { ShrinkableBaseButton } from '@common/components';
 
-interface SplitButtonOption {
+export interface SplitButtonOption<T> {
   label: string;
-  value?: string;
+  value?: T;
   isDisabled?: boolean;
 }
 
-interface SplitButtonProps {
+export interface SplitButtonProps<T> {
   color?: ButtonWithIconProps['color'];
   ariaLabel?: string;
   ariaControlLabel?: string;
-  options: SplitButtonOption[];
-  onClick: (option: SplitButtonOption) => void;
+  options: SplitButtonOption<T>[];
+  onClick: (option: SplitButtonOption<T>) => void;
   Icon?: ButtonWithIconProps['Icon'];
   isDisabled?: boolean;
+  selectedOption: SplitButtonOption<T>;
+  onSelectOption: (option: SplitButtonOption<T>) => void;
 }
 
-export const SplitButton = ({
+export const SplitButton = <T,>({
   color = 'secondary',
   ariaLabel,
   ariaControlLabel,
   options,
   onClick,
   Icon = null,
   isDisabled = false,
-}: SplitButtonProps) => {
+  selectedOption,
+  onSelectOption,
+}: SplitButtonProps<T>) => {
+  console.log(selectedOption);
   const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
-  const [selectedIndex, setSelectedIndex] = React.useState(0);
-
-  const buttonLabel = options[selectedIndex]?.label ?? '';
+  const buttonLabel = selectedOption.label;
```

Comment:
the state change here is not ideal. I was thinking to create a wrapper `StatefulSplitButton` so there is still the option of using the more simple implementation.. 🙄  

Material UI usually allows for their components to be either - i.e. autocomplete - if you pass `undefined` for `value` .. it's a bit of work to implement but not too bad, so could also go that route, though personally I feel like having two explicit components is easier